tasks/8.txt

```diff
@@ -1,3 +1,35 @@
 Create a query to return:
 
-| CategoryName | TotalNumberOfProducts |
\ No newline at end of file
+| CategoryName | TotalNumberOfProducts |
+_________________
+
+db.categories.aggregate([
+    {
+      $lookup:
+        {
+          from: "products",
+          localField: "CategoryID",
+          foreignField: "CategoryID",
+          as: "prdt"
+        }
+    },   
+    {
+        $unwind: "$prdt"
+    },
+    {
+        $group: {
+            _id: "$CategoryName",
+            "TotalNumberOfProducts": {$sum: 1}
+            
+        }
+    },      
+    {
+        $project: {
+            _id: 0,
+            "CategoryName": "$_id",
+            "TotalNumberOfProducts": 1
+        }
+    }
+])
+
```

Comment:
we can do that without unwind and group. See https://docs.mongodb.com/manual/reference/operator/aggregation/size/